lib/serverless_helpers.rb

```diff
@@ -10,13 +10,17 @@ module ServerlessHelpers
   class Error < StandardError; end
 
   extend Dry::Configurable
-  setting(:event_parser, :plain) do |value|
-    SymbolToClassParser.call(
-      argument: value,
-      type: :event,
-      namespace: :parser,
-      base_class: ServerlessHelpers::Event::Parser::Base
-    )
+  setting :event do
+    setting(:parser, :plain) do |value|
+      SymbolToClassParser.call(
+        argument: value,
+        type: :event,
+        namespace: :parser,
+        base_class: ServerlessHelpers::Event::Parser::Base
+      )
+    end
+    setting :provider
+    setting :type
```

Comment:
I tried to do something like this:

```
setting(:type, :plain) do |value|
      SymbolToClassParser.call(
        argument: value,
        type: :event,
        namespace: provider.value,
        base_class: ServerlessHelpers::Event::Base
      )
end
```

But not able to access provider value.